plugin/pkg/response/typify.go

```diff
@@ -28,13 +32,15 @@ const (
 )
 
 var toString = map[Type]string{
-	NoError:    "NOERROR",
-	NameError:  "NXDOMAIN",
-	NoData:     "NODATA",
-	Delegation: "DELEGATION",
-	Meta:       "META",
-	Update:     "UPDATE",
-	OtherError: "OTHERERROR",
+	NoError:             "NOERROR",
+	NameError:           "NXDOMAIN",
+	ServerFailureError:  "SERVFAIL",
```

Comment:
I think it makes sense to lump these together and have a single `ServerError` which for now encompasses both SERVFAIL and NOTIMPL

Reply:
it's hard to do because the `RCode` in dns message do NOT have a `ServerError`. So we need the accurate error type.

Reply:
I want to limit the number of things in this list, OtherError also lumps together errors that are not `Rcode`s. So I still think this makes sense - i.e. a error category that says the server was unresponsive

Reply:
Thanks for your opinion, i have do the modify, please have a review~